Bottom: bcdff8eafdd29e2217a9bb95937904f824deb1cb
Top:    51ed792b0a5713845eb512aebef38e764eb2ac3d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-02-25 00:03:22 +0000

Fix auto-router - we need to actually increment the dir variable!

Was broken accidentally in commit 5f0f788dc12a10a3c01ade7cf00d5ed63922ca47

(Initial C++ compatibility patch)


---

diff --git a/src/autoroute.c b/src/autoroute.c
index b2cbab4..cecf9c0 100644
--- a/src/autoroute.c
+++ b/src/autoroute.c
@@ -2817,7 +2817,7 @@ BreakManyEdges (struct routeone_state * s, rtree_t * targets, rtree_t * tree,
    * in clockwise order, which allows finding corners that can
    * be expanded.
    */
-  for (dir = NORTH; dir <= WEST; directionIncrement(dir))
+  for (dir = NORTH; dir <= WEST; dir = directionIncrement(dir))
     {
       /* don't break the edge we came from */
       if (e->expand_dir != ((dir + 2) % 4))
@@ -2894,7 +2894,7 @@ BreakManyEdges (struct routeone_state * s, rtree_t * targets, rtree_t * tree,
  * moveable as possible.
  */
   first = last = -1;
-  for (dir = NORTH; dir <= WEST; directionIncrement(dir))
+  for (dir = NORTH; dir <= WEST; dir = directionIncrement(dir))
     {
       if (heap[dir] && !heap_is_empty (heap[dir]))
 	{
@@ -3057,7 +3057,7 @@ BreakManyEdges (struct routeone_state * s, rtree_t * targets, rtree_t * tree,
 	}
     }
   /* done with all expansion edges of this box */
-  for (dir = NORTH; dir <= WEST; directionIncrement(dir))
+  for (dir = NORTH; dir <= WEST; dir = directionIncrement(dir))
     {
       if (heap[dir])
 	heap_destroy (&heap[dir]);
